### ISSUE TITLE
polymorphic functions

### ISSUE THREAD

**Issue by bmateus** [gemstation/gemforge#39]
the docs say not to use polymorphic functions, but then how can we implement interfaces that require them? 
For example, in IERC721 we have 

- safeTransferFrom(address,address,uint256)
- safeTransferFrom(address,address,uint256,bytes)

It seem the problem is in the generated LibDiamondHelper where it tries to use .selector 
We get the error 
```
Error (6675): Member "safeTransferFrom" not unique after argument-dependent lookup in type(contract IDiamondProxy).
  --> src/generated/LibDiamondHelper.sol:29:8:
   |
29 | f[0] = IDiamondProxy.safeTransferFrom.selector;
```

however, if instead we specify it as 
`f[9] = bytes4(keccak256(bytes("safeTransferFrom(address,address,uint256,bytes)")));`
then the next step (forge build --names --sizes)  works fine.
I am probably missing something, but wouldn't it be better to modify the template generator to do this instead?

**Comment by pixotchi** [comment]
push

**Comment by hiddentao** [comment]
Great catch. Will take a look at the PR.

**Comment by hiddentao** [comment]
This has now been merged in #42 - thanks!

**Comment by hiddentao** [comment]
2.8.0 is out now.
